internal/dinosaur/pkg/routes/route_loader.go

```diff
@@ -54,6 +54,7 @@ type options struct {
 	AccessControlListMiddleware *acl.AccessControlListMiddleware
 	AccessControlListConfig     *acl.AccessControlListConfig
 	FleetShardAuthZConfig       *auth.FleetShardAuthZConfig
+	RoleAuthZConfig             *auth.RoleAuthZConfig
```

Comment:
Just a thought: Maybe it would be valuable to adjust the name of this variable to indicate that it's used for `admin` routes.
```suggestion
	AdminRoleAuthZConfig             *auth.AdminRoleAuthZConfig
```
That would pull a lot of configuration renaming. Including `pkg/auth/roles_authz.go` and options there, and so on. Middleware itself is ok because it's generic middleware that validates roles.

My main concern is that configurations are starting to be intertwined between admin/internal/customer API. And if you look at command options, it's hard to grasp what is used in the end for specific API authorization. It's not something we should tackle right now, but maybe we should consider some refactoring in the future.

Reply:
I think it's worthwhile to do it within this PR, I've aligned the naming to the `fleetshard-authz` middleware, i.e. re-naming config files + flags.